moteur-server-rest/workflow_manager.py

```diff
@@ -50,17 +88,16 @@ def _kill_workflow(workflow_id, hard_kill):
     moteur_process_class = get_env_variable('MOTEUR_MAIN_CLASS', required=True)
     user_name = get_env_variable('USER', required=True)
     try:
-        command = f"ps -fu {user_name} | grep {moteur_process_class} | grep {workflow_id} | grep -v grep | awk '{{print $2}}' "
-        logger.debug(f"Running command: {command}")
-        result = subprocess.run(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        output = result.stdout.decode('utf-8').strip()
-
-        if output:
-            logger.info(f"Process {output} killed with signal {signal}.")
-            os.system(f"kill -{signal} {output}")
-
-            if hard_kill:
-                update_workflow_status(workflow_id, "Killed")
+        keyword = f"{moteur_process_class}.*{workflow_id}"
+        pids = find_process_pids(keyword, user_name)
```

Comment:
actually, you can use the same pattern as in the other place.
you can also get "moteur_process_class" and "user_name" in "find_process_pids" and only give it the workflow_id as argument.